src/components/CircularCarousel.js

```diff
@@ -230,9 +232,83 @@ export default class CircularCarousel extends Component {
                 </div>
               ))
             }
-
           </div>
+          {isMobile ? countIndicator === 'counter' ? <div
```

Comment:
I'm thinking that the countIndicator and dotIndicator could be separate components. It will reduce the complexity in this component and make them re-usable in other galleries. 
You can pass a current index and total count to them. That way, you won't need to iterate through the array again. Let me know if that makes sense.

Reply:
New commit with components made